hcs-sxc-java-examples/hcs-sxc-java-settlement-demo/src/main/resources/static/dialogs/credits/settle-now-dialog.js

```diff
@@ -70,13 +73,15 @@
                     'Content-Type': 'application/json'
                 },
                 body: postBody
-            }).then(function(response) {
+            }).then(async function(response) {
                 if (response.status===200){
                     showSnackBarMessage("Settlement request sent");
+                    location.hash='settlements';
+                    await renderSettlementsPanel(otherUserId);
```

Comment:
why do we need to block-until this block? The modal closes immediately after any action and the snackbar does asynchronously snow when a response is received. I don't suggest to undo.

Reply:
Unless we wait, when switching to the settlements panel there won't be new data to show. This waits for the pending settlement to be created before switching.